transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -271,15 +282,22 @@ private void validate() {
       // Check if the format specified is valid.
       if (format == null || format.isEmpty()) {
         throw new IllegalArgumentException("Format is not specified. Allowed values are DEFAULT, EXCEL, MYSQL," +
-                                             " RFC4180, PDL & TDF");
+                                             " RFC4180, PDL, TDF or Custom");
       }
 
       // Check if format is one of the allowed types.
       if (!format.equalsIgnoreCase("DEFAULT") && !format.equalsIgnoreCase("EXCEL") &&
         !format.equalsIgnoreCase("MYSQL") && !format.equalsIgnoreCase("RFC4180") &&
-        !format.equalsIgnoreCase("TDF") && !format.equalsIgnoreCase("PDL")) {
+        !format.equalsIgnoreCase("TDF") && !format.equalsIgnoreCase("PDL") && !format.equalsIgnoreCase("Custom")) {
         throw new IllegalArgumentException("Format specified is not one of the allowed values. Allowed values are " +
-                                             "DEFAULT, EXCEL, MYSQL, RFC4180, PDL & TDF");
+                                             "DEFAULT, EXCEL, MYSQL, RFC4180, PDL, TDF or Custom");
+      }
+
+      if (format.equalsIgnoreCase("Custom") && (delimiter == null || delimiter == 0)) {
+        throw new IllegalArgumentException("Please specify the delimiter for format option 'Custom'.");
+      }
+      if (!format.equalsIgnoreCase("Custom") && delimiter != null && delimiter != 0) {
```

Comment:
``delimiter`` is a Character. We are checking delimiter equal to zero to check if it is an empty character , since the default value for each char element is '\u0000' (the null character) which is also equal to 0.